lib/traject/extract_genre.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+require 'traject/extraction_tools'
+extend ExtractionTools
+
+module ExtractGenre
+  def extract_genre
```

Comment:
All of these separate extract modules are very similar. A lot of the methods are repeated across them all and others are so alike that you could create one module that could handle them all by passing in tags and processing blocks methods as arguments. Can you refactor so that we have one module that can handle them all (or at least a majority of them)?

Reply:
@bwatson78 I intentionally wanted to explicitly define the logic pertaining to the validity of a field and how it is extracted in the module I defined for each field. While in this case all fields are mostly tied to `subject` and have similar requirements, I believe our indexing logic across the app should be explicitly defined for each field we do not use `traject` methods to extract, for better readability.

One change I can add for better readability is rename methods like `valid?` and `valid_source?` for each module to be more field specific e.g. `valid_subject_era?` and `valid_subject_era_source?`, what do you think?

Reply:
Ok, I'm going to approve this because you feel strongly about leaving test methods with identical code (like `valid_source?` and `valid_subject_source?`) in each module. I'd take the stance that those identical methods should be named something like `have_approved_language?` knowing that we're applying a language standard across all subject fields.